syntax = "proto3";

package datalink;


message DataLinkRequest {}


message DataLinkData {
  string id = 1;            
  string callsign = 2;      
  string status = 3;        
  double lat = 4;          
  double lon = 5;           
  double velocity = 6;      
  double baroaltitude = 7; 
  double geoaltitude = 8;   
  int64 timestamp = 9;      
}


message DataLinkStreamResponse {
  DataLinkData data = 1;
}


service DataLinkService {
  rpc StreamDataLink (DataLinkRequest) returns (stream DataLinkStreamResponse);
}
